Add UI package + docs (#8)
elastic/azure-vm-extension: build/CONTRIBUTING.md

Azure VM extensions are lightweight applications that handle post-deployment configuration and automation tasks on Azure VMs. Azure hosts many extensions for e.g. VM configuration, monitoring, security, and utility applications. Publishers take an application, then wrap it into an extension, and simplify the installation, so all users need to do is provide mandatory parameters

The main source code is structured in 2 parts:
- handler - contains the windows/linux related scripts that need to install needed dependencies on the vm, download the elastic agent install it and enroll it to Fleet
- packages - contains the UI and miscellaneous installation scripts 

## Handler

The handler implements the Windows Azure Agent defined protocol to integrate with the Azure ecosystem. 

Handler requirements:

> 1. Handler Packaging – The Handler should be packaged as a zip file. This zip package should contain all the binaries related to the handler and HandlerManifest. This package needs to be registered with the Azure image repository. Azure image repository is responsible for managing all versions of all the handlers that are registered with the Azure ecosystem.

Reviewer: ```suggestion
1. Handler Packaging – The handler should be packaged as a zip file. This zip package should contain all the binaries related to the handler and HandlerManifest. This package needs to be registered with the Azure image repository. Azure image repository is responsible for managing all versions of all the handlers that are registered with the Azure ecosystem.
```

> 2. Handler Environment - Handler needs the capability to read the environment file in the format that Azure Agent defines. The environment file defines the locations of various files and folders that the handler needs to use for reading configuration and writing back heartbeat and status.

Reviewer: ```suggestion
2. Handler Environment - The handler needs the capability to read the environment file in the format that Azure Agent defines. The environment file defines the locations of various files and folders that the handler needs to use for reading configuration and writing back heartbeat and status.
```

 

### Handler Package

This is the package that contains the handler binary files and all standard static configuration files. Is is registered with the azure ecosystem and will contain the binaries and the HandlerManifest.json file that is used by the Azure Agent to manage the handler.
 Since we are supporting both Windows and Linux os’s `handler` directory is divided into `windows` and `linux` handler packages for each os version.
Ex. HandlerManifest.json

----
    [{
    "version": 1.0,
    "handlerManifest": {
        "installCommand": "scripts\\windows\\install.cmd",
        "uninstallCommand": "scripts\\windows\\uninstall.cmd",
        "updateCommand": "scripts\\windows\\update.cmd",
        "enableCommand": "scripts\\windows\\enable.cmd",
        "disableCommand": "scripts\\windows\\disable.cmd",
        "rebootAfterInstall": false,
        "reportHeartbeat": false
    }
    }]
----

Elastic Agent Handler Package structure

Binaries will be composed of .cmd, .ps1 and .sh files.

Install command:
- should download and install the necessary dependencies for the elastic agent to be installed.

Enable command:

> - will parse the Ellasticsearch and Kibana urls fromt he cloud ID

Reviewer: ```suggestion
- will parse the Elasticsearch and Kibana urls from the cloud ID
```

> - will ping the ES cluster to retrieve the stack version
> - will download the elastic agent with the selected stack version
> - will install the elastic agent, enroll it to Fleet and start it

Reviewer: ```suggestion
- will ping the ES cluster to retrieve the stack version
- will download the Elastic Agent with the selected stack version
- will install the Elastic Agent, enroll it to Fleet and start it
```

Will also handled cases like update configuration or start vm

Disable command:
 - Should only stop the service (cmd documented in elastic agent docs)

Uninstall command:

> - Should uninstall the service. For deb and rpm, additional effort is made in order to unenroll the agent (api call), stop the agent and remove the elastic agent folders

Reviewer: ```suggestion
- Should uninstall the service. For deb and rpm, additional effort is made in order to unenroll the agent (api call), stop the agent and remove the Elastic Agent folders
```

Update command:

> - Should re-install a new instance of elastic agent

Reviewer: ```suggestion
- Should re-install a new instance of Elastic Agent
```

    "rolename": "<role name for the vm>",
    "instance": "<instance name for the vm>"
    }
    }]
---
`logFolder` - contains the location where the handler should put its log files that might be needed to debug any customer issues. The advantage of putting log files under the folder directed by this location is that these files can be automatically retrieved from the customers VM by using a tool, without actually logging into the VM and copying them over manually. 
`configFolder` - contains the location where the handler will get its configuration settings file. 
`statusFolder` - contains the location where the handler is supposed to write back a file with a structured status of the current state of the work being done by the handler. heartbeatFile - this is the file that is used to communicate the heartbeat of the handler back to the Azure Agent. 
`deploymentid` - this is the deployment id which the vm belongs to. This may change when a VM is captured/backed-up and restored. 
`rolename` -  this is the role name of the vm. This may change when a VM is captured/backed up and restored. 
`instance` - this is the instance name of the vm (can be same as role name)

> ###Handler Configuration

Reviewer: ```suggestion
### Handler Configuration
```
